### ISSUE TITLE
Support different prior for `class Real`

### ISSUE THREAD

**Issue by FrancescoCasalegno** [scikit-optimize/scikit-optimize#603]
The `__init__` method of `class Real` currently supports only two types of prior distribution: `"uniform"` or `"log-uniform"` (see [here](https://github.com/scikit-optimize/scikit-optimize/blob/master/skopt/space/space.py#L183))

However, in many cases we have some prior knowledge about the value of a hyper-parameter. E.g. for the dropout rate in a neural network we may want to have a prior that is bell-shaped and centered around 0.5 (like `scipy.stats.beta(a=2, b=2)`).

`sklearn.model_selection.RandomizedSearchCV` already allows you to specify the sampling distribution of each parameter with `param_distributions=...`, see [here](https://github.com/scikit-learn/scikit-learn/blob/master/sklearn/grid_search.py#L877). 

It would be really nice to have the same possibility of customizing the prior in `scikit-optimize`.

**Comment by betatim** [comment]
Indeed this would be useful. The random number generator used inside a `Dimension` is already a scipy distribution, so allowing users to pass in arbitrary ones would be my suggestion for how to implement this. As opposed to more "some-name-here" strings that get translated.

One thing that will be a bit tricky is that right now `Dimension.rvs()` assumes that the random number generator operates in the "transformed" space. So we can't just pass in a scipy distribution as the user would have to change its parameters when changing the transformation.

**Comment by jolespin** [comment]
@FrancescoCasalegno  Have you made any progress with this?  I could also make good use of this with a very similar situation.  It would be nice to put more mass around 0 and 1 with beta distribution if the parameters are probabilities.
